agent/lib/kontena/service_pods/lifecycle_hook_manager.rb

```diff
@@ -0,0 +1,94 @@
+module Kontena
+  module ServicePods
+    class LifecycleHookManager
+      include Kontena::Helpers::RpcHelper
+      include Common
+
+      attr_reader :service_pod
+
+      # @param [ServicePod] service_pod
+      def initialize(service_pod)
+        @service_pod = service_pod
+      end
+
+      # @return [Boolean]
+      def on_pre_start
+        pre_start_hooks = hooks_for('pre_start')
+        return if pre_start_hooks.size == 0
+
+        pre_pod = service_pod.dup
+        pre_pod.entrypoint = '/bin/sh'
+        pre_pod.cmd = ['-c', 'sleep 600']
+        service_config = config_container(pre_pod)
+        service_container = create_container(service_config)
+        service_container.start!
+        pre_start_hooks.each do |hook|
+          info "running pre_start hook: #{hook['cmd']}"
+          command = ['/w/w', '/bin/sh', '-c', hook['cmd']]
```

Comment:
Now if we prefix each command with `/w/w` we don't need the entrypoint "hacking" for this tmp container?